_posts/2021-09-26-where-the-store-is-stored.md

```diff
@@ -0,0 +1,292 @@
+---
+title: Where is my store stored?
+
+draft: false
+
+page : false
+
+image: /assets/img/where-is-store-stored-illustration.png 
+
+excerpt: How looking at where stores are stored in our Javascript codebases is
+useful to understand the constraints that come with those locations and how the 
+stores can be used.
+
+---
+
+<img
+  alt="Abstract illustration of a person holding two puzzle pieces"
+  src="/assets/img/where-is-store-stored-illustration.png"
+  style="max-height: 480px; margin: 2rem auto"
+/>
+
+Some of us have concepts in mind that they keep returning to over an over, as if
+they haven't completely internalized them. For me as a frontend software
+engineer, it's all about the concepts of *State* and *Store*. I keep looking at
+technical issues through this prism - current and previous colleagues know this
+all too well (sorry colleagues!). I've actually already [written about
+it](/tech/posts/2020-11-22-state-store-in-frontend-codebases) in the past.
+
+I have changed companies a few month ago to join BlaBlaCar and the very cool
+frontend team building and maintaining its web app and find myself in situations
+that make me change how I think about the "frontend Store dilemma" I kept
+referring to. There is now a new question I ask myself again and again looking
+at our frontend codebases:
+
+Where is the store stored?
+
+# State vs Store - one more time 
+
+Let's go back to the difference between the concepts of *State* and *Store*
+even though we sometimes use the two terms indistinctively in frontend
+codebases. In every frontend app, data (the State) is stored somewhere (the
+Store).
+
+When we're thinking about states, we're in the world of ideas. We ask
+questions about what those states represent? Which transition between states are
+valid? How should we shape our states?
+
+When we deal with stores, we're not in the world of ideas anymore. We're back
+in the reality of the code that runs somewhere, mainly in a Javascript runtime.
+We ask where do store out data? How do we read it? How do we write it? How can
```

Comment:
I dont understand that sentence 😅  Did you mean `We ask where do we store our data?` instead ?

Reply:
ahah yup, missing `we` thanks